pkg/roachprod/vm/aws/aws.go

```diff
@@ -566,7 +551,7 @@ func (p *Provider) Create(
 	}
 
 	var zones []string // contains an az corresponding to each entry in names
-	if !opts.GeoDistributed && (useDefaultZones || len(expandedZones) == 1) {
```

Comment:
Methinks we still want the right disjunct, `len(expandedZones) == 1`, no? Otherwise, `opts.CreateZones` is ignored.

Reply:
> Otherwise, opts.CreateZones is ignored.

Why is that? If I pass a single AZ, it _will_ run the `else` block but that will just trivially "distribute" every node to the requested AZ, I believe.

This code is rather difficult to follow. If I'm reading it right, I think the only thing that actually matters here is `opts.GeoDistributed`. If `--geo` is passed, we run the zone placement algorithm. Otherwise, we pick a random AZ in the first region (since we already set `expandedZones` to `defaultZones` above).

Reply:
>  If I'm reading it right, I think the only thing that actually matters here is opts.GeoDistributed.

Yep, you're right. The guard can be simplified further to `!opts.GeoDistributed`. `len(expandedZones) >= 1` is established before the if-stmt.